src/core/qgslegendrenderer.cpp

```diff
@@ -661,7 +676,18 @@ QSizeF QgsLegendRenderer::drawGroup( const LegendComponentGroup &group, QgsRende
           currentY += mSettings.style( s ).margin( QgsLegendStyle::Top );
         }
         QSizeF groupSize;
-        groupSize = drawGroupTitle( groupItem, context, columnContext, currentY );
+        ColumnContext columnContextForItem = columnContext;
+        double indentWidth =  component.indent;
+        if ( s == QgsLegendStyle::Subgroup )
+        {
+          indentWidth -= mSettings.style( QgsLegendStyle::Subgroup ).indent( );
```

Comment:
why are these negative values, when the similar code for drawLayerTitle uses positive values?

Reply:
Negative value used here because we need to undo the indentation for the group/subgroup title (only the group items should be indented). I've added comments to the code to avoid confusion